### PR TITLE
add Haliburton e1031 platform

### PR DESCRIPTION
**Pull request by l2okii**
**- What I did**
- port SONiC for Haliburton platform.

**- How I did it**
- add platform module haliburton.

**- How to verify it**
- It must has Halibuton platform in broadcom platform.

**- Description for the changelog**

- add Haliburton e1031 platform

**- A picture of a cute animal (not mandatory but encouraged)**
cat

### CODE REVIEW

File: device/celestica/x86_64-cel_e1031-r0/E1031-T48S4/port_config.ini [hunk 1]
```diff
@@ -1,53 +1,5 @@
...
-Ethernet47      48      1000
-Ethernet48      54     10000
-Ethernet49      53     10000
-Ethernet50      56     10000
-Ethernet51      55     10000
+# name          lanes                 alias	index
+Ethernet-SFP-1       49           Eth1		14
+Ethernet-SFP-2       50           Eth2		15
+Ethernet-SFP-3       51           Eth3		16
+Ethernet-SFP-4       52           Eth4		17
```
Comment (lguohan) [inline device/celestica/x86_64-cel_e1031-r0/E1031-T48S4/port_config.ini]: why do you remove all 1g ports? Can you keep the origin format? also I do not understand why you change lane information on 10G sfp ports.

File: src/sonic-config-engine/sonic_platform.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ def get_machine_info():
             tokens = line.split('=')
             if len(tokens) < 2:
                 continue
-            machine_vars[tokens[0]] = tokens[1].strip()
+            machine_vars[tokens[0]] = tokens[1].strip().strip("'")
```
Comment (lguohan) [inline src/sonic-config-engine/sonic_platform.py:24]: do not modify this file.

File: installer/x86_64/install.sh [hunk 3]
```diff
@@ -65,9 +65,9 @@ fi
 echo "onie_platform: $onie_platform"
 
 # default console settings
-CONSOLE_PORT=0x3f8
-CONSOLE_DEV=0
-CONSOLE_SPEED=9600
+CONSOLE_PORT=0x2f8
+CONSOLE_DEV=1
+CONSOLE_SPEED=115200
```
Comment (lguohan) [inline installer/x86_64/install.sh:70]: please do not modify this file.

File: device/celestica/x86_64-cel_e1031-r0/plugins/sfputil.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    def qsfp_ports(self):
+        return range(0, self.PORTS_IN_BLOCK + 1)
+
+    @property
+    def port_to_eeprom_mapping(self):
+        return self._port_to_eeprom_mapping
+
+    def __init__(self):
+        # Override port_to_eeprom_mapping for class initialization
+        eeprom_path = '/sys/bus/i2c/devices/i2c-{0}/{0}-0050/eeprom'
```
Comment (lguohan) [inline device/celestica/x86_64-cel_e1031-r0/plugins/sfputil.py]: the eeprom driver only works for qsfp module, need to use oom module to support sfp module. 
oom module has been added to sonic kernel, so you can use it directly.

more information can be found at.
https://github.com/opencomputeproject/oom

**Comment by lguohan** [comment]
where is fancontrol module? how to dynamically adjust fan speed?

File: device/celestica/x86_64-cel_e1031-r0/E1031-T48S4/port_config.ini (hunk 1 above)
Comment (l2okii) [reply device/celestica/x86_64-cel_e1031-r0/E1031-T48S4/port_config.ini]: OK, i will bring 1g port back.
For the lanes, i reference from R0882-M1011-01- Rev1.0 Haliburton HW Design Spec.pdf

image

File: device/celestica/x86_64-cel_e1031-r0/plugins/sfputil.py (hunk 4 above)
Comment (mudsut4ke) [reply device/celestica/x86_64-cel_e1031-r0/plugins/sfputil.py]: OK I will use oom driver to support qsfp and sfp module

File: installer/x86_64/install.sh (hunk 3 above)
Comment (mudsut4ke) [reply installer/x86_64/install.sh:70]: OK, but we found the issue that we can't access SONiC console after we install it with this old config

File: src/sonic-config-engine/sonic_platform.py (hunk 2 above)
Comment (mudsut4ke) [reply src/sonic-config-engine/sonic_platform.py:24]: OK, but we found this method return machine name with the single quote mark

**Comment by mudsut4ke** [comment]
@lguohan please review my latest commit

**Comment by lguohan** [comment]
incorporated into the stretch branch. close this pr.

**Pull request closed without merging**